### PR TITLE
Fix `isExternal` for falsy values.

### PR DESCRIPTION
**Pull request by sgomes**
Add an early return for non-string falsy values (the empty string can be processed normally).

While the typed method doesn't expect falsy values, the nature of JavaScript is that they can be passed in.

#### Changes proposed in this Pull Request

* Handle falsy values correctly in `isExternal`.

#### Testing instructions

The changes should be pretty innocuous, but it may be interesting to consult #100768-z, which this PR should fix. The live branch should make it easy to test the changes.

### CODE REVIEW

File: client/lib/url/is-external.ts [hunk 1]
```diff
@@ -11,10 +11,16 @@ import { isLegacyRoute } from 'lib/route/legacy-routes';
...
-const BASE_HOSTNAME = 'base.invalid';
+const BASE_HOSTNAME = '__domain__.invalid';
 const BASE_URL = `http://${ BASE_HOSTNAME }`;
 
-export default function isExternal( url: TypedURL ): boolean {
+export default function isExternal( url: URLString ): boolean {
+	// While TypeScript should ensure that `url` really is a string, this method
+	// is still used in a lot of JavaScript contexts, without type checks.
+	if ( ! url && url !== '' ) {
+		return true;
```
Comment (sgomes) [inline client/lib/url/is-external.ts]: Note: this matches the previous behaviour of this method.

**Comment by shantanu2704** [comment]
Hi there! Just following up here to see what we can do to get this merged in. This was a result of a client reporting that they were unable to GA tracking as they were presented with a blank(white) screen on the analytics page. I'm unable to switch to the user and test on the live branch, so if there's a way to do that, please let me know! 

Thank you!

**Comment by sgomes** [comment]
> Hi there! Just following up here to see what we can do to get this merged in. This was a result of a client reporting that they were unable to GA tracking as they were presented with a blank(white) screen on the analytics page.

Hey @shantanu2704! Just waiting on a review. I'll ping a few folks to see if I can get that sped up!

@jsnajdr @blowery @sirreal Any of you got a few minutes to take a look at this bugfix PR? Thanks!

> I'm unable to switch to the user and test on the live branch, so if there's a way to do that, please let me know!

Would anyone know if there happens to be a way of impersonating a user on a live branch?

In any case, this fix should be harmless (although I strongly expect it to). It may not fix the issue, but it shouldn't introduce any problems either.

File: client/lib/url/is-external.ts [hunk 2]
```diff
@@ -11,10 +11,16 @@ import { isLegacyRoute } from 'lib/route/legacy-routes';
 
 // Base URL used for URL parsing. The WHATWG URL API doesn't support relative
 // URLs, so we always need to provide a base of some sort.
-const BASE_HOSTNAME = 'base.invalid';
+const BASE_HOSTNAME = '__domain__.invalid';
```
Comment (jsnajdr) [inline client/lib/url/is-external.ts]: How is this change related to the fix?

Comment (sgomes) [reply client/lib/url/is-external.ts]: It's not, it must have slipped in from another branch. I'll remove it.

Comment (sgomes) [reply client/lib/url/is-external.ts]: Done.

File: client/lib/url/is-external.ts (hunk 1 above)
Comment (sgomes) [reply client/lib/url/is-external.ts]: I just pulled out the old code to double-check and yes, that's correct.

I did notice that invalid URLs were throwing exceptions (and they didn't in the original code, interestingly), so I added a `try`/`catch` to match the original behaviour and return `false`.

**Review by jsnajdr — APPROVED** [review]
:ship:

**Comment by sgomes** [comment]
Thanks for the review, @jsnajdr ! 👍

**Comment by sgomes** [comment]
@shantanu2704 This is now live in production. Let me know if the issue hasn't been fixed!

**Comment by shantanu2704** [comment]
Hey @sgomes ! This is now working as expected. Thank you!

**Comment by sgomes** [comment]
> Hey @sgomes ! This is now working as expected. Thank you!

Perfect! Thank you for letting me know, @shantanu2704!
